core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+    # + code_evaluation.ResultsInError (cannot be 100% recovered)
+    # - code_evaluation.ErrorContains
+    # + coord_two_dim.Within
+    # + coord_two_dim.NotWithin
+    # - graph.HasGraphProperty
+    # - graph.IsIsomorphicTo
+    # + math_expression.IsMathematicallyEquivalentTo
+    # + music_phrase.Equals
+    # + music_phrase.IsLongerThan
+    # + music_phrase.HasLengthInclusivelyBetween
+    # + music_phrase.IsEqualToExceptFor
+    # + music_phrase.IsTranspositionOf
+    # + music_phrase.IsTranspositionOfExceptFor
+    # + nonnegative_int.Equals
+    # + normalized_string.Equals
+    # + normalized_string.CaseSensitiveEquals
+    # + normalized_string.StartsWith
+    # + normalized_string.Contains
+    # + normalized_string.FuzzyEquals
+    # + real.Equals
+    # + real.IsLessThan
+    # + real.IsGreaterThan
+    # + real.IsLessThanOrEqualTo
+    # + real.IsGreaterThanOrEqualTo
+    # + real.IsInclusivelyBetween
+    # + real.IsWithinTolerance
+    # + set_of_html_string.Equals
+    # + set_of_html_string.ContainsAtLeastOneOf
+    # + set_of_html_string.DoesNotContainAtLeastOneOf
+    # + set_of_unicode_string.Equals
+    # + set_of_unicode_string.IsSubsetOf
+    # + set_of_unicode_string.IsSupersetOf
+    # + set_of_unicode_string.HasElementsIn
+    # + set_of_unicode_string.HasElementsNotIn
+    # + set_of_unicode_string.OmitsElementsIn
+    # + set_of_unicode_string.IsDisjointFrom
+
+    # NOTE TO DEVELOPERS: This was never a modifiable value, so it will always
+    # take the minimum value. It wasn't stored in answers, but it does not need
+    # to be reconstituted.
+    _NOTE_DURATION_FRACTION_PART = 1
+
+    # A modern version of
+    # https://github.com/oppia/oppia/blob/94df0f/extensions/widgets/interactive/FileReadInput/FileReadInput.py.
+    class FileReadInput(base.BaseInteraction):
+        """Interaction for uploading a file."""
+
+        name = 'File upload'
+        description = 'A widget for uploading files.'
+        display_mode = base.DISPLAY_MODE_INLINE
+        _dependency_ids = []
+        answer_type = 'UnicodeString'
+        instructions = None
+        narrow_instructions = None
+        needs_summary = False
+        _customization_arg_specs = []
+
+    # A modern version of
+    # https://github.com/oppia/oppia/blob/94df0f/extensions/widgets/interactive/TarFileReadInput/TarFileReadInput.py.
+    class TarFileReadInput(base.BaseInteraction):
+        """Interaction for uploading a file."""
+
+        name = 'Tar file upload'
+        description = 'A widget for uploading tar files.'
+        display_mode = base.DISPLAY_MODE_INLINE
+        _dependency_ids = []
+        answer_type = 'TarFileString'
+        instructions = None
+        narrow_instructions = None
+        needs_summary = False
+        _customization_arg_specs = []
+
+    @classmethod
+    def _ensure_removed_interactions_are_in_registry(cls):
+        """Checks if the FileReadInput or TarFileReadInput deleted interactions
```

Comment:
Not possible to purge, no. We could do a job after the migration to remove these from the new storage model, which is effectively the same net outcome. Purging these require doing the complex matching of the answer to an exploration version which is the most complex part of the migration job; I'd rather not try getting it to work in a second job especially since I've already validated these answers are being migrated successfully.